### PR TITLE
Fix PositionsAppender#build for null Block

### PR DESCRIPTION
**Pull request by lukasz-stec**
## Description
If the created block had only null positions
the dedicated PositionAppender's were
not correctly reset.

> Is this change a fix, improvement, new feature, refactoring, or other?

Fix
> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)
core query engine, PartitionedOutputOperator
> How would you describe this change to a non-technical end user or system administrator?

bug fix
## Related issues, pull requests, and links

Fixes https://github.com/trinodb/trino/issues/12667

## Documentation

( ) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( ) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: core/trino-main/src/test/java/io/trino/operator/output/TestPagePartitioner.java [hunk 1]
```diff
@@ -392,37 +417,48 @@ public static Object[][] partitioningMode()
...
-                .flatMap(type -> Stream.of(PartitioningMode.values())
-                        .map(partitioningMode -> new Object[] {type, partitioningMode}))
+        return Arrays.stream(getTypesWithNullBlock())
+                .flatMap(typeWithNullBlock -> Stream.of(PartitioningMode.values())
+                        .map(partitioningMode -> new Object[] {typeWithNullBlock[0], partitioningMode}))
+                .toArray(Object[][]::new);
+    }
+
+    @DataProvider(name = "typesWithPartitioningModeAndNullBlock")
+    public static Object[][] typesWithPartitioningModeAndNullBlock()
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/operator/output/TestPagePartitioner.java:435]: `typesWithPartitioningModeAndNullBlock` -> `typesWithNullBlockAndPartitioningMode`

**Review by sopel39 — APPROVED** [review]

File: core/trino-main/src/test/java/io/trino/operator/output/TestPagePartitioner.java [hunk 2]
```diff
@@ -191,6 +199,23 @@ public void testOutputForMultipleSimplePages(PartitioningMode partitioningMode)
         assertThat(partitioned).containsExactlyInAnyOrderElementsOf(expected); // order is different due to 2 partitions joined
     }
 
+    @Test(dataProvider = "typesWithPartitioningModeAndNullBlock")
+    public void testOutputForMultipleNullPages(Type type, Block nullBlock, PartitioningMode partitioningMode)
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/operator/output/TestPagePartitioner.java:203]: this should be in `TestPositionsAppender`

File: core/trino-main/src/test/java/io/trino/operator/output/TestPagePartitioner.java [hunk 3]
```diff
@@ -191,6 +199,23 @@ public void testOutputForMultipleSimplePages(PartitioningMode partitioningMode)
         assertThat(partitioned).containsExactlyInAnyOrderElementsOf(expected); // order is different due to 2 partitions joined
     }
 
+    @Test(dataProvider = "typesWithPartitioningModeAndNullBlock")
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/operator/output/TestPagePartitioner.java:202]: Think about other edge cases too:
1. Empty block
2. Rle followed by empty block
3. Empty rle followed by null block
Can be separate PR too

File: core/trino-main/src/test/java/io/trino/operator/output/TestPagePartitioner.java [hunk 4]
```diff
@@ -191,6 +199,23 @@ public void testOutputForMultipleSimplePages(PartitioningMode partitioningMode)
...
+                1,
+                createLongSequenceBlock(0, 1), // partition block
+                nullBlock);
+
+        // process twice to force flush
+        processPages(pagePartitioner, partitioningMode, page);
+        processPages(pagePartitioner, partitioningMode, page);
+
+        List<Object> partitioned = readChannel(outputBuffer.getEnqueuedDeserialized(), 1, type);
+        assertThat(partitioned).containsExactlyInAnyOrderElementsOf(Arrays.asList(null, null)); // order is different due to 2 partitions joined
```
[the remark this answers is not in the document]
Comment (lukasz-stec) [reply core/trino-main/src/test/java/io/trino/operator/output/TestPagePartitioner.java:216]: this is related to the `containsExactlyInAnyOrderElementsOf` but its dropped anyway

File: core/trino-main/src/test/java/io/trino/operator/output/TestPagePartitioner.java (hunk 1 above)
Comment (lukasz-stec) [reply core/trino-main/src/test/java/io/trino/operator/output/TestPagePartitioner.java]: removed

File: core/trino-main/src/test/java/io/trino/operator/output/TestPagePartitioner.java (hunk 3 above)
Comment (lukasz-stec) [reply core/trino-main/src/test/java/io/trino/operator/output/TestPagePartitioner.java]: Since build must reset the appenders state I added these cases in one go in `io.trino.operator.output.TestPositionsAppender#testConsecutiveBuilds`

**Review by lukasz-stec — COMMENTED** [review]
comments addressed

**Review by sopel39 — APPROVED** [review]
thanks!
